### PR TITLE
AEIM-1057 - Add logrotate to unison

### PR DESCRIPTION
**Pull request by daaang**
Since this is using a new module, I'd like to test it in an environment before merging into production, to make sure the module does what we want.

### CODE REVIEW

**Comment by daaang** [comment]
Debian's default cron.daily/logrotate looks like this:

```bash
test -x /usr/sbin/logrotate || exit 0
/usr/sbin/logrotate /etc/logrotate.conf
```

Puppet replaces it with this one:

```bash
OUTPUT=$(/usr/sbin/logrotate /etc/logrotate.conf 2>&1)
EXITVALUE=$?
if [ $EXITVALUE != 0 ]; then
    /usr/bin/logger -t logrotate "ALERT exited abnormally with [$EXITVALUE]"
    echo "${OUTPUT}"
fi
exit $EXITVALUE
```

Which, while different, doesn't offend me. So unless anyone objects, I'll just let puppet do its thing in this case.

**Comment by daaang** [comment]
Precisely, these are the four files that get created or modified:

/etc/logrotate.conf:

```
# THIS FILE IS AUTOMATICALLY DISTRIBUTED BY PUPPET.
# ANY CHANGES WILL BE OVERWRITTEN.

create
rotate 4
weekly

# configurable file rotations
include /etc/logrotate.d
```

/etc/logrotate.d/debian_btmp:

```
# THIS FILE IS AUTOMATICALLY DISTRIBUTED BY PUPPET.  ANY CHANGES WILL BE
# OVERWRITTEN.

/var/log/btmp {
  create 0660 root utmp
  missingok
  monthly
  rotate 1
}
```

/etc/logrotate.d/debian_wtmp:

```
# THIS FILE IS AUTOMATICALLY DISTRIBUTED BY PUPPET.  ANY CHANGES WILL BE
# OVERWRITTEN.

/var/log/wtmp {
  create 0664 root utmp
  missingok
  monthly
  rotate 1
}
```

/etc/logrotate.d/unison:

```
# THIS FILE IS AUTOMATICALLY DISTRIBUTED BY PUPPET.  ANY CHANGES WILL BE
# OVERWRITTEN.

/var/log/unison.log {
  compress
  daily
  delaycompress
  missingok
  notifempty
  rotate 7
}
```

File: manifests/profile/unison.pp [hunk 1]
```diff
@@ -12,6 +12,17 @@
   Array $servers = [],
   Array $clients = []
 ) {
+  include nebula::profile::logrotate
+
+  logrotate::rule { 'unison':
+    path          => '/var/log/unison.log',
```
Comment (aelkiss) [inline manifests/profile/unison.pp]: `/var/log/unison.log` won't be sufficient; the client logs to `/var/log/unison_<%= @title %>.log`. I believe wildcards can be specified in the logrotate config (that's what we're currently doing for apache), so I think just changing it to `/var/log/unison*.log` should be fine.

Comment (daaang) [reply manifests/profile/unison.pp]: I've done exactly that, so this should be ready to merge.

**Pull request merged**